src/GenericCyclotomics.jl

```diff
@@ -365,13 +365,6 @@ function (R::GenericCycloRing)(f::Dict{UPolyFrac,UPoly}; simplify::Bool=true)  #
     return GenericCyclo(f, R)
   end
 
-  # congruence preparation
-  if R.congruence !== nothing
-    q = gen(base_ring(R), 1)
-    substitute = R.congruence[2] * q + R.congruence[1]
-    substitute_inv = (q - R.congruence[1]) * 1//R.congruence[2]
-  end
-
   # reduce numerators modulo denominators
   L = NTuple{4,UPoly}[]
   for (g, c) in f
```

Comment:
Style / taste question, but: I like to avoid nested control flow, esp. multi-level, to improve code clarity. So here I'd do
```suggestion
  for (g, c) in f
    is_zero(c) && continue
```
and then drop the `if !iszero(c)` and thereby reduce the indention level inside the loop